lunatrace/bsl/backend/src/config/load-environment-vars.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import dotenv from 'dotenv';
+import { z } from 'zod';
+
+import { commonEnvVarKeys } from './common-env-vars';
+
+dotenv.config({ path: process.env.NODE_ENV === 'production' ? '.env' : '.env.dev' });
+
+const missingEnvVars: string[] = [];
+
+type EnvVar<S extends z.Schema> = {
+  varKey: string;
+  backupVarKey?: string;
+  defaultValue?: z.infer<S>;
+  castTo: S;
+};
+
+// Takes an env var key object from "common-env-vars.ts", reads it from the environment, and returns a scalar or throws if it cannot
+function checkEnvVar<E extends EnvVar<z.Schema>>(varConf: E): z.infer<E['castTo']> {
```

Comment:
This is incredibly confusing

Reply:
can you elaborate on why you find this “incredibly confusing”?

Reply:
I mean, i agree with alex. This is a small amount of really confusing code. 

In the future, we should look first for libraries when we go to do things like this. It will save us a lot of time.